source/extensions/transport_sockets/tls/ocsp/ocsp.cc

```diff
@@ -0,0 +1,307 @@
+  //    successful            (0),  -- Response has valid confirmations
+  //    malformedRequest      (1),  -- Illegal confirmation request
+  //    internalError         (2),  -- Internal error in issuer
+  //    tryLater              (3),  -- Try again later
+  //                                -- (4) is not used
+  //    sigRequired           (5),  -- Must sign the request
+  //    unauthorized          (6)   -- Request unauthorized
+  // }
+  CBS status;
+  if (!CBS_get_asn1(&cbs, &status, CBS_ASN1_ENUMERATED)) {
+    throw EnvoyException("OCSP ResponseStatus is not a well-formed ASN.1 ENUMERATED");
+  }
+
+  auto status_ordinal = *CBS_data(&status);
+  switch (status_ordinal) {
+  case 0:
+    return OcspResponseStatus::Successful;
+  case 1:
+    return OcspResponseStatus::MalformedRequest;
+  case 2:
+    return OcspResponseStatus::InternalError;
+  case 3:
+    return OcspResponseStatus::TryLater;
+  case 5:
+    return OcspResponseStatus::SigRequired;
+  case 6:
+    return OcspResponseStatus::Unauthorized;
+  default:
+    throw EnvoyException(absl::StrCat("Unknown OCSP Response Status variant: ", status_ordinal));
+  }
+}
+
+ResponsePtr Asn1OcspUtility::parseResponseBytes(CBS& cbs) {
+  // ResponseBytes ::=  SEQUENCE {
+  //     responseType        RESPONSE.
+  //                             &id ({ResponseSet}),
+  //     response            OCTET STRING (CONTAINING RESPONSE.
+  //                             &Type({ResponseSet}{@responseType}))
+  // }
+  CBS elem, response;
+  if (!CBS_get_asn1(&cbs, &elem, CBS_ASN1_SEQUENCE)) {
+    throw EnvoyException("OCSP ResponseBytes is not a well-formed SEQUENCE");
+  }
+
+  auto oid_str = Asn1Utility::parseOid(elem);
+  if (!CBS_get_asn1(&elem, &response, CBS_ASN1_OCTETSTRING)) {
+    throw EnvoyException("Expected ASN.1 OCTETSTRING for response");
+  }
+
+  if (oid_str == BasicOcspResponse::OID) {
+    return Asn1OcspUtility::parseBasicOcspResponse(response);
+  }
+  throw EnvoyException(absl::StrCat("Unknown OCSP Response type with OID: ", oid_str));
+}
+
+std::unique_ptr<BasicOcspResponse> Asn1OcspUtility::parseBasicOcspResponse(CBS& cbs) {
+  // BasicOCSPResponse       ::= SEQUENCE {
+  //    tbsResponseData      ResponseData,
+  //    signatureAlgorithm   AlgorithmIdentifier{SIGNATURE-ALGORITHM,
+  //                             {sa-dsaWithSHA1 | sa-rsaWithSHA1 |
+  //                                  sa-rsaWithMD5 | sa-rsaWithMD2, ...}},
+  //    signature            BIT STRING,
+  //    certs            [0] EXPLICIT SEQUENCE OF Certificate OPTIONAL
+  // }
+  CBS elem;
+  if (!CBS_get_asn1(&cbs, &elem, CBS_ASN1_SEQUENCE)) {
+    throw EnvoyException("OCSP BasicOCSPResponse is not a wellf-formed ASN.1 SEQUENCE");
+  }
+  auto response_data = Asn1OcspUtility::parseResponseData(elem);
+  auto signature_alg = Asn1Utility::parseAlgorithmIdentifier(elem);
+  auto signature = Asn1Utility::parseBitString(elem);
+  // TODO(daniel-goldstein): Verify this signature
```

Comment:
This part seems critically important. Can this be fixed before it's submitted? If not, please rename the function to include a warning like `parseBasicOcspResponseUnsafeDoNotUse()`.

Reply:
On a second look at this, I'm realizing we don't currently have access to the responder's public key to actually verify this signature. It would need to be provided in the config along with the ocsp response. I can clarify this dependency in the comment.